### DISCUSSION TITLE
1.4.1

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#313] [Обновления]
## Список изменений
- Добавлено 39 новых языков перевода меню ([#297](https://github.com/ilyhalight/voice-over-translation/pull/297))
- Добавлено 2 новых языка устанавливаемых как язык видео (Латышский и Литовский - [#297](https://github.com/ilyhalight/voice-over-translation/pull/297))
- Добавлено 38 новых языков для TTS (`az`, `sq`, `am`, `hy`, `af`, `eu`, `my`, `bg`, `bs`, `cy`, `hu`, `gl`, `el`, `zu`, `kn`, `ca`, `km`, `lo`, `mk`, `ml`, `mt`, `mn`, `ne`, `nl`, `pa`, `ro`, `sr`, `si`, `sk`, `sl`, `sw`, `su`, `hr`, `et`, `da`, `fi`, `uz`, `pl`, `sv` - [#297](https://github.com/ilyhalight/voice-over-translation/pull/297))
- Улучшено форматирование ссылок на YouTube, теперь, язык должен определяться точнее ([#297](https://github.com/ilyhalight/voice-over-translation/pull/297))
- Добавлена поддержка TikTok

  Работает только в видео типа: /@НИК/video/АЙДИ_ВИДЕО (не работает в ленте)

- Добавлена поддержка Udemy

  **Если у курса нету субтитров на вашем языке, то перевод не будет выполнен.**
  **Перевод очень далек от перевода в том же ютубе, но это всё равно лучше чем ничего.**

  Работает, только, для добавленных курсов т.е. соответственно вам нужно быть авторизованным в аккаунт Udemy. Для работы перевода необходимо через меню расширения установить accessToken. Он необходим для получения ссылки на файлы субтитров и видео, без которых Яндекс не переводит данный сервис.

  Гайд по установке accessToken: **[тык](https://github.com/ilyhalight/voice-over-translation/wiki/%5BRU%5D-Where-to-get-Udemy-Access-Token%3F)**

- Добавлена поддержка Coursera

  **Если у курса нету субтитров на вашем языке, то перевод не будет выполнен.**
  **Перевод очень далек от перевода в том же ютубе, но это всё равно лучше чем ничего.**

  Работает только в режиме прохождения курсов, т.е. вы должны авторизоваться в аккаунте перейти в "Мои курсы" и выбрать нужный курс.

- Добавлена поддержка ProxyTok (протестировано на proxitok.pabloferreiro.es и proxitok.privacydev.net. Ограничения теже, что и для обычного TikTok)
- Добавлена поддержка нового встраиваемого плеера Bilibili (player.bilibili.com)
- Исправлена ошибка из-за которой нельзя было выбрать языковую пару en-ru в mail.ru видео и rutube ([#298](https://github.com/ilyhalight/voice-over-translation/issues/298))
- Фикс [#285](https://github.com/ilyhalight/voice-over-translation/issues/285), [#283](https://github.com/ilyhalight/voice-over-translation/issues/283), [#282](https://github.com/ilyhalight/voice-over-translation/issues/282) ([#297](https://github.com/ilyhalight/voice-over-translation/pull/297))
- В protobuf обновлен тип для translationHelp
- Преобразование языка в формат "хх" перенесено в отдельную функцию langTo6391
- Для OrangeMonkey добавлено уведомление о необходимости использовать Cloudflare-версию
- Исправлена ошибка из-за которой мог установиться не поддерживаемый язык ([#292](https://github.com/ilyhalight/voice-over-translation/discussions/292))
- Исправлена ошибка из-за которой язык видео мог определяться в формате отличном от xx
- Улучшена логика работы обновлений indexedDB
- Возвращен пункт меню "Проксировать полученное аудио"
- Исправлено изменение процента громкости видео при изменение громкости через слайдер ютуба
- Переработано получение и установка громкости видео
- Исправлен сброс установленной громкости звука при открытие нового видео
- Локализация была разделена на отдельные файлы и перенесена на Github
- vot-cli был перенесен в отдельный **[репозиторий](https://github.com/FOSWLY/vot-cli)**
- Удалена папка с web-server (уже давно не используется)

## What's Changed (autogenerated)
* Add localization provider by @MrSoczekXD in https://github.com/ilyhalight/voice-over-translation/pull/275
* README.md: Add support iOS for UserScripts by @ACherepkov1989 in https://github.com/ilyhalight/voice-over-translation/pull/280
* Some fixes by @MrSoczekXD in https://github.com/ilyhalight/voice-over-translation/pull/288
* Улучшения форматирования текста для определения языка by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/297
* [1.4.1] dev -> master by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/312

## New Contributors
* @ilyhalight made their first contribution in https://github.com/ilyhalight/voice-over-translation/pull/312

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.4.0.3...1.4.1

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.4.1'>1.4.1</a>.</em>

**Comment by programist2304** [comment]
Сколько доступно голосов? И как можно поменять голос?

**Reply by NullVerdict** [reply]
Нельзя менять голоса, они выбираются автоматически Яндексом.

**Comment by Tynch05** [comment]
Приветствую, а можно ли настроить так, чтобы автоматически видео определялись как английские и перевод на русский. В моем мире mail-ru скрипт определяет видео как русский по умолчанию и переводит на английский. Приходится до нажатия на кнопку "перевести" выбирать что видео на английском и перевод на русский. Использую хром, tampermonkey.
![01](https://github.com/ilyhalight/voice-over-translation/assets/33279455/7f7ce56d-048b-47db-8ac9-aa5e81676b5e)

**Reply by ilyhalight** [reply]
в 872 строчке поменяй на en и собери расширение
![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/c4d7e3ea-58e8-464a-a5ee-61a83a3f5631)

**Reply by Tynch05** [reply]
Спасибо, получилось. А можно ли как-нибудь удалить выпадающий список перевода субтитров. Чтобы вообще её не было, может удалить какой-нибудь участок кода?

**Reply by ilyhalight** [reply]
Ищите такую строчку в index.js и после нее добавляете `select.hidden = true`

![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/297b638d-808e-4cf9-a73d-c94e9aed236d)

**Comment by infowater** [comment]
Друзья! 
Может ли данный скрипт делать перевод с русского на другие языки? 
Например на немецкий или английский. 
И еще один вопрос. 
Как можно переводить сайты с помощью данного скрипта на других сайтах типа Seasonvar?

**Reply by ilyhalight** [reply]
> На 04.11.2023 большая часть языков доступных для выбора в качестве озвучки были отключены на стороне Яндекса. Мне неизвестно будут ли они возвращены позднее или нет. Из тех, на которые точно можно выполнить перевод остались: Русский, Английский, Казахский.

>  Как можно переводить сайты с помощью данного скрипта на других сайтах типа Seasonvar?

Только пытаться добавить поддержку самостоятельно

**Reply by infowater** [reply]
Если бы кто-то показал как это сделать, я бы занялся и выложил свой шедевр. К сожалению я не программист :(
